File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationRecipeSaveViewHolder.java
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class RecipeManipulationRecipeSaveViewHolder extends RecyclerView.ViewHolder {

    private final OnSaveRecipeClickedListener onSaveRecipeClickedListener;

    public RecipeManipulationRecipeSaveViewHolder(ViewGroup viewGroup,
                                                  OnSaveRecipeClickedListener onSaveRecipeClickedListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.viewholder_recipe_save, viewGroup, false));
        ButterKnife.bind(this, itemView);

        this.onSaveRecipeClickedListener = onSaveRecipeClickedListener;
    }

    public void bind(RecipeManipulationViewItemWrapper item) { }

    @OnClick(R.id.add_recipe) public void onSaveRecipeClicked() {
        onSaveRecipeClickedListener.onSaveRecipeClicked();
    }

    public interface OnSaveRecipeClickedListener {
        void onSaveRecipeClicked();
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationRecipeSaveViewHolder.kt
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.viewholder_recipe_save.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class RecipeManipulationRecipeSaveViewHolder(override val containerView: View,
                                             private val onSaveRecipeClickedListener: OnSaveRecipeClickedListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(item: RecipeManipulationViewItemWrapper) {
        setupClickListeners()
    }

    private fun setupClickListeners() {
        add_recipe.setOnClickListener { onSaveRecipeClickedListener.onSaveRecipeClicked() }
    }

    interface OnSaveRecipeClickedListener {
        fun onSaveRecipeClicked()
    }

}
